### PR TITLE
4 new Application Tests added

### PR DESCRIPTION
**Pull request by andrew3737**
liquidRouterOutputAll verifies that liquid routers output form the other three sides that the input came from.
sorterOutputAll verifies that wanted item goes straight through the sorter whereas the unwanted items are sorted out the sides of the sorter.
routerOutputAll verifies that routers output form the other three sides that the input came from.
junctionOutputAll verifies that items can be passed on conveyors under and over each other through a junction without being mixed.

### CODE REVIEW

File: tests/src/test/java/ApplicationTests.java [hunk 1]
```diff
@@ -290,6 +290,120 @@ void liquidJunctionOutput(){
...
+        router.setBlock(Blocks.liquidRouter, Team.sharded);
+        conduitLeft.setBlock(Blocks.conduit, Team.sharded,2);
+        conduitUp2.setBlock(Blocks.conduit, Team.sharded, 1);
+        conduitRight.setBlock(Blocks.conduit, Team.sharded, 0);
+        leftTank.setBlock(Blocks.liquidTank, Team.sharded);
+        topTank.setBlock(Blocks.liquidTank, Team.sharded);
+        rightTank.setBlock(Blocks.liquidTank, Team.sharded);
+
+        updateBlocks(200);
+        assertEquals(true, rightTank.build.liquids.currentAmount() > 0);
```
Comment (LeoSko) [inline tests/src/test/java/ApplicationTests.java:313]: Why not simply `assertTrue`?

Comment (andrew3737) [reply tests/src/test/java/ApplicationTests.java:313]: Anuken requested to use assert equals for comparisons

Comment (andrew3737) [reply tests/src/test/java/ApplicationTests.java:313]: assertTrue is more refined in this case though, I'm going to make an adjustment here.

Comment (andrew3737) [reply tests/src/test/java/ApplicationTests.java]: Let me know what you think of the adjustment.

Comment (LeoSko) [reply tests/src/test/java/ApplicationTests.java]: I'd just wait for Anuken to see. The change is too minor to think much about.

File: tests/src/test/java/ApplicationTests.java [hunk 2]
```diff
@@ -290,6 +290,119 @@ void liquidJunctionOutput(){
...
+        junction.setBlock(Blocks.junction, Team.sharded);
+
+
+        vault1.setBlock(Blocks.vault, Team.sharded);
+        vault2.setBlock(Blocks.vault, Team.sharded);
+
+        updateBlocks(200);
+        assertEquals(Items.coal, vault1.build.items.first());
+        assertEquals(Items.copper, vault2.build.items.first());
+        assertNotEquals(Items.copper, vault1.build.items.first());
```
Comment (Anuken) [inline tests/src/test/java/ApplicationTests.java:402]: These two `assertNotEquals` are redundant. If the items are already known to be coal and copper from the above assertions, they can't possibly be anything else.

File: tests/src/test/java/ApplicationTests.java [hunk 3]
```diff
@@ -290,6 +290,119 @@ void liquidJunctionOutput(){
...
+        source1.build.configureAny(Items.coal);
+        source2.setBlock(Blocks.itemSource, Team.sharded);
+        source2.build.configureAny(Items.copper);
+        conveyor1.setBlock(Blocks.conveyor, Team.sharded, 1);
+        conveyor2.setBlock(Blocks.conveyor, Team.sharded, 2);
+        conveyor3.setBlock(Blocks.conveyor, Team.sharded, 1);
+        conveyor4.setBlock(Blocks.conveyor, Team.sharded, 2);
+
+        junction.setBlock(Blocks.junction, Team.sharded);
+
```
Comment (Anuken) [inline tests/src/test/java/ApplicationTests.java:394]: unnecessary newline?

File: tests/src/test/java/ApplicationTests.java (hunk 2 above)
Comment (andrew3737) [reply tests/src/test/java/ApplicationTests.java:402]: I will make some updates here

File: tests/src/test/java/ApplicationTests.java (hunk 3 above)
Comment (andrew3737) [reply tests/src/test/java/ApplicationTests.java:394]: I'll clear that up for you!

Comment (andrew3737) [reply tests/src/test/java/ApplicationTests.java:394]: I'll clear that up for you!

File: tests/src/test/java/ApplicationTests.java (hunk 2 above)
Comment (andrew3737) [reply tests/src/test/java/ApplicationTests.java:402]: I see what you mean, I will update the tests.

**Comment by andrew3737** [comment]
I updated the files in this most recent commit. I change the names of two of the tests to better reflect what their intention is, as well as removed the extra newline and redundant assertEquals statements.

**Pull request merged**